### PR TITLE
Move GGD unit tests to separate file

### PR DESCRIPTION
**Pull request by muneebahmed10**
Description
-----------

Moves GGD tests that do not need the network to a separate file.

The following unit tests were moved from `aws_test_greengrass_discovery.c` to `aws_test_ggd_unit.c` with no changes to the tests themselves, besides naming:

-    GetIPOnInterface
-    CheckForContentLengthString
-    Jsoneq
-    CheckMatch
-    GetCertificate
-    GetCore
-    IsIPvalid

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+#define ggdTestJSON_GROUPID_TOKEN_INDEX    4
+#define ggdTestJSON_CORE_TOKEN_INDEX       9
+#define ggdTestJSON_PORT_ADDRESS_1         1234
+#define ggdTestJSON_PORT_ADDRESS_3         4321
+
+#define ggdJSON_FILE_GROUPID               "GGGroupId"
+#define ggdJSON_FILE_THING_ARN             "thingArn"
+#define ggdJSON_FILE_HOST_ADDRESS          "HostAddress"
+#define ggdJSON_FILE_CERTIFICATE           "CAs"
+#define ggdJSON_FILE_PORT_NUMBER           "PortNumber"
```
Comment (aggarw13) [inline libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: These macros don't seem to be used in the file. Can you make sure to prune any resr of the macros that are not used in the tests?

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+    RUN_TEST_CASE( GGD_Unit, GetCertificate );
+    RUN_TEST_CASE( GGD_Unit, GetCore );
+    RUN_TEST_CASE( GGD_Unit, IsIPvalid );
+}
+
+TEST( GGD_Unit, IsIPvalid )
+{
+    BaseType_t xStatus;
+
+    if( TEST_PROTECT() )
```
Comment (aggarw13) [inline libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: What is the use of `TEST_PROTECT()` in these unit tests? Is there any multi-threaded behavior in these tests that need this macro?

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c (hunk 1 above)
Comment (muneebahmed10) [reply libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: Good catch, will remove

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c (hunk 2 above)
Comment (muneebahmed10) [reply libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: I'm unsure what the purpose of these are. These tests were just copied over from aws_test_greengrass_discovery.c with no changes, as I don't think making any code changes to the tests falls under this PR's scope

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c [hunk 3]
```diff
@@ -0,0 +1,586 @@
+/*
+ * FreeRTOS Greengrass V2.0.1
```
Comment (aggarw13) [inline libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: The file naming seems inconsistent for this new file (`aws_tests_ggd_unit.c`) and the existing file `aws_tests_greengrass_discovery.c`. Can you update them to be consistent and also use the `_system` suffix in the old tests' file?

File: libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c [hunk 4]
```diff
@@ -164,32 +141,6 @@ TEST( Full_GGD, JSONRequestAbort )
...
-        /** @}*/
-    }
-    else
-    {
-        TEST_FAIL();
-    }
-}
-
 
 TEST( Full_GGD, GetGGCIPandCertificate )
```
Comment (aggarw13) [inline libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c]: _Full__GGD seems misleading when considering the whole suite of tests for Greengrass. Using `GDD_System` would be appropriate now

Comment (muneebahmed10) [reply libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c]: Changed

File: libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c (hunk 3 above)
Comment (muneebahmed10) [reply libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_unit.c]: Updated

**Pull request merged**